src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -469,46 +478,51 @@ protected Field GetOrCreateField(string name, string value, LazyStringValue lazy
 
         private IEnumerable<AbstractField> GetOrCreateNumericField(IndexField field, object value, Field.Store storage, Field.TermVector termVector = Field.TermVector.NO)
         {
-            var fieldName = field.Name + Constants.Documents.Indexing.Fields.RangeFieldSuffix;
+            var fieldNameDouble = field.Name + Constants.Documents.Indexing.Fields.RangeFieldSuffixDouble;
+            var fieldNameLong = field.Name + Constants.Documents.Indexing.Fields.RangeFieldSuffixLong;
 
-            var cacheKey = new FieldCacheKey(field.Name, null, storage, termVector,
-                _multipleItemsSameFieldCount.ToArray());
+            var multipleItemsSameFieldCountArray = _multipleItemsSameFieldCount.ToArray();
+            var cacheKeyDouble = new FieldCacheKey(fieldNameDouble, null, storage, termVector, multipleItemsSameFieldCountArray);
+            var cacheKeyLong = new FieldCacheKey(fieldNameLong, null, storage, termVector, multipleItemsSameFieldCountArray);
 
-            NumericField numericField;
-            CachedFieldItem<NumericField> cached;
-
-            if (_numericFieldsCache.TryGetValue(cacheKey, out cached) == false)
-            {
-                _numericFieldsCache[cacheKey] = cached = new CachedFieldItem<NumericField>
-                {
-                    Field = numericField = new NumericField(CreateFieldName(fieldName), storage, true)
-                };
-            }
-            else
-            {
-                numericField = cached.Field;
-            }
+            var numericFieldDouble = GetNumericFieldFromCache(cacheKeyDouble, storage);
+            var numericFieldLong = GetNumericFieldFromCache(cacheKeyLong, storage);
 
             double doubleValue;
             long longValue;
 
             switch (BlittableNumber.Parse(value, out doubleValue, out longValue))
             {
                 case NumberParseResult.Double:
-                    if (field.SortOption == SortOptions.NumericLong)
-                        yield return numericField.SetLongValue((long) doubleValue);
-                    else
-                        yield return numericField.SetDoubleValue(doubleValue);
+                    yield return numericFieldDouble.SetDoubleValue(doubleValue);
+                    yield return numericFieldLong.SetLongValue((long)doubleValue);
```

Comment:
Why are we outputting both double & long values here? It should be one or the other, no?